packages/swapper/src/swappers/thorchain/getThorTradeQuote/getTradeQuote.ts

```diff
@@ -87,23 +63,97 @@ export const getThorTradeQuote = async ({
     const priceRatio = await getPriceRatio(deps, { sellAssetId, buyAssetId })
     const rate = bnOrZero(1).div(priceRatio).toString()
     const buyAmount = normalizeAmount(bnOrZero(sellAmount).times(rate))
-    const sellAssetTradeFee = bnOrZero(feeData.tradeFee).times(bnOrZero(feeAssetRatio))
 
+    const tradeFee = await estimateTradeFee(deps, buyAsset.assetId)
+    const sellAssetTradeFee = bnOrZero(tradeFee).times(bnOrZero(feeAssetRatio))
     // padding minimum by 1.5 the trade fee to avoid thorchain "not enough to cover fee" errors.
     const minimum = fromBaseUnit(sellAssetTradeFee.times(1.5).toString(), sellAsset.precision)
 
-    return {
+    const commonQuoteFields: CommonQuoteFields = {
```

Comment:
can just use an `Omit` over the dupe `CommonQuoteFields` type
```suggestion
    const commonQuoteFields: Omit<TradeQuote<ChainId>, 'allowanceContract' | 'feeData'> = {```